spec/secret_code_spec.rb

```diff
@@ -0,0 +1,81 @@
+require '../lib/secret_code'
+
+RSpec.describe SecretCode do
+  describe '#initialize' do
+    xit 'generates a secrect code when initialized' do
+      # mock to check that gen_secret_code is called
+      # but would that be testing behavior or simply ruby?
+    end
+  end
+
+  describe '#gen_secret_code' do
+    context 'creates a code that' do
+      code = gen_secret_code
+      it 'is 4 characters' do
+        expect(code.length).to eq 4
+      end
+      it 'only contains numbers 1-6' do
+        expect(code.all? { |val| val.to_i.between?(1, 6) }).to eq true
+      end
+    end
+  end
+
+  # not sure if i should be testing this?
```

Comment:
If it's public, it should be tested

Reply:
it's not, it's a private method that is called in a public method that is tested below it. So it's technically already being tested?, but i'm not sure whether I should test is as well for more specific-ness?

Reply:
If it's private, it can't be tested. You can test the public method thoroughly

Reply:
Or extract it out into another file and then test it

Reply:
Right I can't even access it because it's private *facepalm*

That makes sense, I think that I'll just end up removing it as it's really only testing if this method can count correctly x_x